enterprise/decorators.py

```diff
@@ -134,11 +140,15 @@ def wrapper(request, *args, **kwargs):
         # Enterprise-linked IdP and the pipeline will get them back here.
         if not request.user.is_authenticated():
             parsed_current_url = urlparse(request.get_full_path())
-            parsed_query_string = parse_qs(parsed_current_url.query)
-            parsed_query_string.update({
+            parsed_query_string = parse_qsl(parsed_current_url.query)
+
+            additional_query_parameters = {
                 'tpa_hint': enterprise_customer.identity_provider,
                 FRESH_LOGIN_PARAMETER: 'yes'
-            })
+            }
+            for query_parameter_key, query_parameter_value in iteritems(additional_query_parameters):
```

Comment:
Not just `key` and `value` would be nice instead of `query_parameter_key/value` ?

Reply:
I think that verbose variable names make the statement more readable.

Reply:
agree but For local variables that are only relevant for a short number of lines, having long verbose names actually makes them less efficient :)